File: assignment1/prime.java
import java.util.*;

public class prime {
    private int[] PrimeTable;

    public prime(){ }
    public static void main(String[] args) {
        Scanner sc1 = new Scanner(System.in);
        Scanner sc2 = new Scanner(System.in);

        int inputNum;
        String inputString;
        String[] result;
        ArrayList<Integer> numList;
        boolean InRange = true;
        while (true){
            System.out.print(">> Input the number of numbers to process: ");
            inputNum = sc1.nextInt();

            if(inputNum < 2 || inputNum > 30){
                System.out.println("Input count range should be  \"1 < [INPUT] < 31\" \n");
                continue;
            }
            System.out.print(">> Input the numbers to be processed:\n");

            inputString = sc2.nextLine();
            result = inputString.split(" ");
            if(result.length != inputNum) {
                System.out.println("Input count Error!");
                continue;
            }


            numList = new ArrayList();
            for(int i=0; i<inputNum; i++){
                numList.add(Integer.parseInt(result[i]));
            }


            for(int i=0; i<numList.size(); i++){
                if(numList.get(i) < 1 || numList.get(i) > 100000){
                    InRange = false;
                    break;
                }
            }

            if(!InRange){
                System.out.println("Range : 1 ~ 100,000");
                InRange = true;
                numList.clear();
                continue;
            }

            long startTime = System.nanoTime();
            HashSet<Integer> uniqueSet = new HashSet<>(numList);
            ArrayList<Integer> uniqueList = new ArrayList<>(uniqueSet);
            Collections.sort(uniqueList); // uniqueList : already sorted


            int GCD = uniqueList.get(0);
            prime pr = new prime();
            for(int i=0; i<uniqueList.size() - 1; i++){
                GCD = pr.get_GCD(GCD, uniqueList.get(i));
            }
            System.out.println("GCD of input number is " + GCD + "\n");

            pr.PrimeTable = new int[uniqueList.get(uniqueList.size()-1) + 1];
            for(int i=0; i<uniqueList.size()-1; i++)
                pr.count_Prime(uniqueList.get(i), uniqueList.get(i+1));

            long endTime = System.nanoTime();
            double diffTime = (endTime - startTime) / 1000000000.0;
            System.out.println("Total execution time using Java is " + diffTime + " seconds!");
            break;


        }
    }
    int get_GCD(int a, int b){
        int n;
        while(a!=0){
            n = b%a;
            b = a;
            a = n;
        }
        return b;
    }
    void count_Prime(int min, int max){
        int cnt = 0;

        for(int i=2; i<=max; i++){
            PrimeTable[i] = 1;
        }

        for(int i=2; i<=max; i++){
            if(PrimeTable[i] == 0)
                continue;
            for(int j=2*i; j<=max; j+=i)
                PrimeTable[j] = 0;
        }

        for(int i=min; i<=max; i++){
            if(PrimeTable[i] != 0)
                cnt++;
        }

        System.out.println("Number of prime numbers between " + min + ", " + max + ": " + cnt);
    }
}

File: assignment1/prime.py
import time

def get_GCD(a, b):
    while(a!=0) :
        n = b%a
        b = a
        a = n
    return b

def count_Prime(min, max):
    cnt = 0
    PrimeTable = list();
    for i in range(0, max+1):
        PrimeTable.append(1)
        
    for i in range(2, max+1):
        if(PrimeTable[i] == 0):
            continue
        for j in range(2*i, max+1, i):
            PrimeTable[j] = 0
    
    for i in range(min, max+1):
        if(PrimeTable[i] != 0):
            cnt = cnt + 1
    
    print("Number of prime numbers between {}, {}: {}".format(min, max, cnt))
    return;


while True:
    print(">> Input the number of numbers to process:", end=' ');
    inputNum = int(input());
    
    if(inputNum < 2 or inputNum > 30) :
        print("Input count range should be \"1 < [INPUT] < 31\" ")
        continue;
    
    print(">> Input the numbers to be processed:")
    numList = list(map(int, input().split()))
    if(len(numList) != inputNum):
        print("Input Count shoud be {}".format(inputNum))
        continue
    
    uniqueList = set(numList)
    uniqueList = list(uniqueList)
    uniqueList.sort()

    InRange = True
    for i in range(len(uniqueList)):
        if(uniqueList[i] < 1 or uniqueList[i] > 100000):
            InRange = False
            break
    
    if(InRange == False):
        print("Range : 1 ~ 100,000")
        InRange = True
        continue
    
    startTime = time.time()

    GCD = uniqueList[0]
    for i in range(len(uniqueList) - 1):
        GCD = get_GCD(GCD, uniqueList[i])

    print("GCD of input number is {}".format(GCD))
    
    for i in range(len(uniqueList) -1):
        count_Prime(uniqueList[i], uniqueList[i+1])
        
    endTime = time.time()
    execTime = endTime - startTime
    
    print("\nTotal execution time using Python is {} seconds!".format(execTime))
    break

    